Make TensorImplPtr custom deleter copyable.

shoumikhin opened the pull request:
Summary: .

Differential Revision: D62338334

kirklandsign commented on extension/tensor/tensor_impl_ptr.cpp:
```diff
@@ -30,20 +30,23 @@ static void noop_deleter(void*) {}
+  // C++20, so any data it holds must be copyable too. Hence, we use shared_ptr
+  // to hold the data and metadata to avoid unnecessary copies.
+  std::shared_ptr<void> data;
+  std::shared_ptr<std::vector<exec_aten::SizesType>> sizes;
+  std::shared_ptr<std::vector<exec_aten::DimOrderType>> dim_order;
+  std::shared_ptr<std::vector<exec_aten::StridesType>> strides;
```
Curious, for these do we need std::shared_ptr?

shoumikhin committed:
Make TensorImplPtr custom deleter copyable. (#5161)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/5161

.

Reviewed By: kirklandsign

Differential Revision: D62338334
